### PR TITLE
Add conversion from arb to Integer and fmpz

### PR DESCRIPTION
**Pull request by albinahlback**
Please let me know if this can be optimized.

Need to add tests as well as add `convert(::Type{T}, ::arb)`.

### CODE REVIEW

**Comment by thofma** [comment]
We usually do not add `convert` methods. Also, I think it would be better to just call `unique_integer`.

**Comment by albinahlback** [comment]
Let me know if you want me to remove or adjust something.

File: src/arb/arb.jl [hunk 1]
```diff
@@ -135,6 +133,35 @@ function convert(::Type{Float64}, x::arb)
...
 
+@doc Markdown.doc"""
+    fmpz(x::arb)
+
+Return $x$ as an `fmpz` if it represents an unique integer, else throws an
+error.
+"""
+function fmpz(x::arb)
+  (b, n) = unique_integer(x)
+  b == 0 ? error("Argument must represent be an unique integer") : return n
```
Comment (thofma) [inline src/arb/arb.jl:144]:
```suggestion
  !b ? error("Argument must represent be an unique integer") : return n
```
I think `b` is just a boolean.

**Comment by thofma** [comment]
I meant that we don't usually add `convert` methods and we should not start with it. The function `convert` is called in lots of places where one usually does not want it to appear, most notable for `getindex/setindex` and `haskey`.

**Comment by albinahlback** [comment]
@thofma Let me know what you think.

File: src/arb/arb.jl [hunk 2]
```diff
@@ -135,6 +133,25 @@ function convert(::Type{Float64}, x::arb)
...
 
+@doc Markdown.doc"""
+    fmpz(x::arb)
+
+Return $x$ as an `fmpz` if it represents an unique integer, else throws an
+error.
+"""
+function fmpz(x::arb)
+  (b, n) = unique_integer(x)
+  !b ? error("Argument must represent be an unique integer") : return n
```
Comment (thofma) [inline src/arb/arb.jl:144]:
```suggestion
  !isexact(x) && error("Argument must represent a unique integer")
  (b, n) = unique_integer(x)
  !b ? error("Argument must represent a unique integer") : return n
```

**Comment by thofma** [comment]
I just noticed that my comment about just using `unique_integer` was not quite correct, sorry. I corrected it.

**Comment by albinahlback** [comment]
@thofma do you like clicking buttons? ;-)

**Comment by wbhart** [comment]
He's just retriggering tests. It's the easiest way.

**Comment by thofma** [comment]
I know you also get messaged, but the other way is like three clicks more!

**Comment by wbhart** [comment]
{counts the number of letters he just pressed to type that} :-)

**Pull request merged**